python/nwm_client_new/src/hydrotools/nwm_client/NWMFileProcessor.py

```diff
@@ -0,0 +1,125 @@
+"""
+==================
+NWM File Processor
+==================
+Tools for processing NWM data in NetCDF (.nc) format.
+
+Classes
+-------
+NWMFileProcessor
+"""
+from pathlib import Path
+import xarray as xr
+import dask.dataframe as dd
+import pandas as pd
+import numpy as np
+import numpy.typing as npt
+from typing import List, Union
+
+class NWMFileProcessor:
+    """Provides an interface for methods used to process National Water Model 
+    data from NetCDF (.nc) format to xarray.Dataset, dask.dataframe.Dataframe, 
+    or pandas.DataFrame.
+    """
+
+    @classmethod
+    def get_dataset(
+        cls,
+        input_directory: Union[str, Path],
+        feature_id_filter: npt.ArrayLike = [],
+        variables: List[str] = ["reference_time", "time", "streamflow"]
+        ) -> xr.Dataset:
+        """Generate an xarray.Dataset from an input directory of NWM .nc files.
+
+        Parameters
+        ----------
+        input_directory: str, pathlib.Path, required
+            Directory containing collection of National Water Model NetCDF 
+            files.
+        feature_id_filter: array-like, optional, default []
+            Subset of feature IDs to return.
+        variables: list of str, optional, default ["reference_time", "time",
+             "streamflow"]
+             List of variables to retrieve from source files. Options include: 
+             'time', 'reference_time', 'feature_id', 'crs', 'streamflow', 
+             'nudge', 'velocity', 'qSfcLatRunoff', 'qBucket', 'qBtmVertRunoff'
+
+        Returns
+        -------
+        xarray.Dataset of input_directory lazily loaded.
+        """
+        # Resolve input directory
+        input_directory = Path(input_directory)
+
+        # Generate file list
+        file_list = [f for f in input_directory.glob("*.nc")]
+
+        # Open dataset
+        ds = xr.open_mfdataset(file_list, engine="netcdf4")
+
+        # Prepare feature_id filter
+        if len(feature_id_filter) != 0:
+            # Convert to array
+            feature_id_filter = np.asarray(feature_id_filter)
+
+            # Validate filter IDs
+            check = np.isin(feature_id_filter, ds.feature_id)
```

Comment:
Is it worth simply `try`ing to select the filter and then catching a `KeyError` before doing the `check = np.isin(...)`.  This way you WARN about which ID's were not selectable